samples/BoltOn.Samples.Application/Handlers/GetAllGradesHandler.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using BoltOn.Mediator.Pipeline;
+using BoltOn.Samples.Application.Abstractions.Data;
+using BoltOn.Samples.Domain.Entities;
+
+namespace BoltOn.Samples.Application.Handlers
+{
+    public class GetAllGradesRequest : IQuery<IEnumerable<Grade>>
+    {
+        public int StudentId { get; set; }
+    }
+
+    public class GetAllGradesHandler : IRequestAsyncHandler<GetAllGradesRequest, IEnumerable<Grade>>
+    {
+        private readonly IGradeRepository _gradeRepository;
+
+        public GetAllGradesHandler(IGradeRepository gradeRepository)
+        {
+            _gradeRepository = gradeRepository;
+        }
+
+        public async Task<IEnumerable<Grade>> HandleAsync(GetAllGradesRequest request, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var grade1 = new Grade { Id = Guid.NewGuid().ToString(), StudentId = 1, CourseName = "physics", Score = "A-", Year = 2017 };
+            var grade2 = new Grade { Id = Guid.NewGuid().ToString(), StudentId = 2, CourseName = "computers", Score = "A-", Year = 2018 };
+
+            var add = _gradeRepository.Add(grade1);
```

Comment:
why not you create integration tests instead of all the methods here? something like this https://github.com/gokulm/BoltOn/blob/master/tests/BoltOn.Tests/Data/EF/RepositoryTests.cs

Reply:
@gokulm : will do it in next PR